File: CloudFormation/testSrc/com/intellij/aws/cloudformation/tests/MappingSecondLevelKeyReferenceResolveTests.java
package com.intellij.aws.cloudformation.tests;

import com.intellij.aws.cloudformation.references.CloudFormationMappingSecondLevelKeyReference;

public class MappingSecondLevelKeyReferenceResolveTests extends ResolveTestsBase {
  public MappingSecondLevelKeyReferenceResolveTests() {
    super(CloudFormationMappingSecondLevelKeyReference.class);
  }

  public void testFindInMapRefToTopLevelKey() throws Exception {
    assertEntityResolve("findInMapRefToSecondLevelKey", "Arch");
  }

  @Override
  protected String getTestDataPath() {
    return TestUtil.getTestDataPath("/secondLevelKeyResolve/");
  }
}

File: CloudFormation/src/test/kotlin/com/intellij/aws/cloudformation/tests/MappingSecondLevelKeyReferenceResolveTests.kt
package com.intellij.aws.cloudformation.tests

import com.intellij.aws.cloudformation.references.CloudFormationMappingSecondLevelKeyReference
import java.io.File

class MappingSecondLevelKeyReferenceResolveTests : ResolveTestsBase(CloudFormationMappingSecondLevelKeyReference::class.java) {
  fun testFindInMapRefToSecondLevelKey() {
    assertEntityResolve("findInMapRefToSecondLevelKey.template", "Arch")
  }

  override val testDataRoot: File
    get() = TestUtil.getTestDataFile("secondLevelKeyResolve")
}
